Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerFileImpl.cs

```diff
@@ -0,0 +1,150 @@
+
+        internal const string CredentialCacheFileName = "temporary_credential.json";
+
+        private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<SFCredentialManagerFileImpl>();
+
+        private readonly string _jsonCacheDirectory;
+
+        private readonly string _jsonCacheFilePath;
+
+        private readonly FileOperations _fileOperations;
+
+        private readonly DirectoryOperations _directoryOperations;
+
+        private readonly UnixOperations _unixOperations;
+
+        private readonly EnvironmentOperations _environmentOperations;
+
+        public static readonly SFCredentialManagerFileImpl Instance = new SFCredentialManagerFileImpl(FileOperations.Instance, DirectoryOperations.Instance, UnixOperations.Instance, EnvironmentOperations.Instance);
+
+        internal SFCredentialManagerFileImpl(FileOperations fileOperations, DirectoryOperations directoryOperations, UnixOperations unixOperations, EnvironmentOperations environmentOperations)
+        {
+            _fileOperations = fileOperations;
+            _directoryOperations = directoryOperations;
+            _unixOperations = unixOperations;
+            _environmentOperations = environmentOperations;
+            SetCredentialCachePath(ref _jsonCacheDirectory, ref _jsonCacheFilePath);
+        }
+
+        private void SetCredentialCachePath(ref string _jsonCacheDirectory, ref string _jsonCacheFilePath)
+        {
+            var customDirectory = _environmentOperations.GetEnvironmentVariable(CredentialCacheDirectoryEnvironmentName);
+            _jsonCacheDirectory = string.IsNullOrEmpty(customDirectory) ? Path.Combine(HomeDirectoryProvider.HomeDirectory(_environmentOperations), CredentialCacheDirName) : customDirectory;
+            if (!_directoryOperations.Exists(_jsonCacheDirectory))
+            {
+                _directoryOperations.CreateDirectory(_jsonCacheDirectory);
+            }
+            _jsonCacheFilePath = Path.Combine(_jsonCacheDirectory, CredentialCacheFileName);
+            s_logger.Info($"Setting the json credential cache path to {_jsonCacheFilePath}");
+        }
+
+        internal void WriteToJsonFile(string content)
+        {
+            s_logger.Debug($"Writing credentials to json file in {_jsonCacheFilePath}");
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                _fileOperations.Write(_jsonCacheFilePath, content);
+            }
+            else
+            {
+                if (!_directoryOperations.Exists(_jsonCacheDirectory))
+                {
+                    _directoryOperations.CreateDirectory(_jsonCacheDirectory);
+                }
+                s_logger.Info($"Creating the json file for credential cache in {_jsonCacheFilePath}");
+                if (_fileOperations.Exists(_jsonCacheFilePath))
+                {
+                    s_logger.Info($"The existing json file for credential cache in {_jsonCacheFilePath} will be overwritten");
```

Comment:
I review the uses of the WriteToJsonFile and validate that we are rewriting the file but including the old content. We should override the JSON file because we have to preserve the JSON file structure.